src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -300,12 +323,148 @@ def finalize(self) -> RankBasedMetricResults:  # noqa: D102
             metrics=self.metrics,
             rank_and_candidates=_iter_ranks(ranks=self.ranks, num_candidates=self.num_candidates),
         )
+        if not self.clear_on_finalize:
+            return result
+
         # Clear buffers
         self.ranks.clear()
         self.num_candidates.clear()
 
         return result
 
+    def finalize_multi(self, num: int = 1_000, seed: int = 42) -> Mapping[str, Sequence[float]]:
+        """Bootstrap from :meth:`finalize`.
+
+        :param num:
+            the number of resampling steps
+        :param seed:
+            the random seed.
+
+        :return:
+            a flat dictionary from metric names to list of values
+        """
+        result: DefaultDict[str, List[float]] = defaultdict(list)
+
+        for i in range(num):
+            generator = numpy.random.default_rng(seed=seed + i)
+
+            def resample(
+                entry: Tuple[ExtendedTarget, RankType, np.ndarray, np.ndarray, Optional[np.ndarray]],
+                generator=generator,
```

Comment:
why pass this as an argument? you can also use `nonlocal` if there are scope issues

Reply:
https://github.com/pykeen/pykeen/pull/1084/commits/36c3cf25eb5396600c7eb8399e600ac038f26e2e